editoast/src/views/infra/objects.rs

```diff
@@ -0,0 +1,103 @@
+use crate::api_error::ApiError;
+use crate::{api_error::ApiResult, db_connection::DBConnection, schema::ObjectType};
+use diesel::sql_types::{Array, Integer, Jsonb, Text};
+use diesel::{sql_query, QueryableByName, RunQueryDsl};
+use rocket::http::Status;
+use rocket::response::status::Custom;
+use rocket::serde::json::{json, Error as JsonError, Json, Value as JsonValue};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, thiserror::Error)]
+enum GetObjectsErrors {
+    #[error("No object ids provided")]
+    NoObjectIdsProvided,
+}
+
+impl ApiError for GetObjectsErrors {
+    fn get_status(&self) -> Status {
+        match self {
+            GetObjectsErrors::NoObjectIdsProvided => Status::BadRequest,
+        }
+    }
+
+    fn get_type(&self) -> &'static str {
+        match self {
+            GetObjectsErrors::NoObjectIdsProvided => "editoast:infra:objects:NoObjectIdsProvided",
+        }
+    }
+}
+
+#[derive(QueryableByName, Debug, Clone, Serialize, Deserialize)]
+pub struct Test {
+    #[sql_type = "Jsonb"]
+    pub railjson: JsonValue,
+    #[sql_type = "Array<Jsonb>"]
+    geometry: Vec<JsonValue>,
+}
+/// Return the railjson list of a specific OSRD object
+#[get("/<infra>/objects/<object_type>", data = "<obj_ids>")]
+pub async fn get_objects(
+    infra: i32,
+    object_type: ObjectType,
+    obj_ids: Result<Json<Vec<String>>, JsonError<'_>>,
+    conn: DBConnection,
+) -> ApiResult<Custom<JsonValue>> {
```

Comment:
You must handle few more cases:

- When there are duplicates in `obj_ids`
- When the object doesn't have geometry (both `schematic` and `geographic` fields should be set to `null`)
- When we call this endpoint with `SwitchType`, currently the function fails (you can do a special query for that).

Reply:
Thanks for the review. I tried something for those different cases, I don't know if it suits you.